castai/resource_castai_eks_cluster.go

```diff
@@ -139,44 +147,55 @@ func resourceCastaiEksClusterRead(ctx context.Context, data *schema.ResourceData
 
 	log.Printf("[INFO] Getting cluster information.")
 
-	response, err := client.ExternalClusterAPIGetClusterWithResponse(ctx, data.Id())
+	resp, err := client.ExternalClusterAPIGetClusterWithResponse(ctx, data.Id())
 	if err != nil {
 		return diag.FromErr(err)
-	} else if response.StatusCode() == http.StatusNotFound {
+	} else if resp.StatusCode() == http.StatusNotFound {
 		log.Printf("[WARN] Removing cluster %s from state because it no longer exists in CAST.AI", data.Id())
 		data.SetId("")
 		return nil
 	}
 
-	data.Set(FieldEksClusterCredentialsId, *response.JSON200.CredentialsId)
+	data.Set(FieldEKSClusterCredentialsId, *resp.JSON200.CredentialsId)
 
-	if response.JSON200.Eks != nil {
-		data.Set(FieldEksClusterAccountId, *response.JSON200.Eks.AccountId)
-		data.Set(FieldEksClusterRegion, *response.JSON200.Eks.Region)
-		data.Set(FieldEksClusterName, *response.JSON200.Eks.ClusterName)
-		data.Set(FieldEksClusterInstanceProfileArn, *response.JSON200.Eks.InstanceProfileArn)
+	if resp.JSON200.Eks != nil {
+		data.Set(FieldEKSClusterAccountId, *resp.JSON200.Eks.AccountId)
+		data.Set(FieldEKSClusterRegion, *resp.JSON200.Eks.Region)
+		data.Set(FieldEKSClusterName, *resp.JSON200.Eks.ClusterName)
+		data.Set(FieldEKSClusterInstanceProfileArn, *resp.JSON200.Eks.InstanceProfileArn)
 	}
 
-	agentToken, err := retrieveAgentToken(ctx, client)
-	if err != nil {
-		return diag.FromErr(err)
+	if _, ok := data.GetOk(FieldEKSClusterAgentToken); !ok {
```

Comment:
I'm probably missing something, but why still retrieve the Agent Token if it's switched out for Cluster token in the examples? When would we need it?

Reply:
I kept it to not introduce breaking change, we should remove it with next major version release